### PR TITLE
Software cleanup

### PR DESCRIPTION
**Pull request by codemime**
Remove redundant JSON fields and unused software code. Follows #15843.

### CODE REVIEW

File: data/json/items/software.json [hunk 1]
```diff
@@ -2,91 +2,51 @@
     {
         "type":"GENERIC",
         "id": "software_useless",
-        "symbol": " ",
```
Comment (mugling) [inline data/json/items/software.json]: This isn't the current default value as set in `itype.h`

Comment (codemime) [inline data/json/items/software.json]: It doesn't matter. It shall never be shown.

File: gfx/ChestHole16Tileset/tile_config.json [hunk 2]
```diff
@@ -14538,36 +14538,6 @@
           "rotates": false
         },
         {
-          "id": "software_medical",
```
Comment (mugling) [inline gfx/ChestHole16Tileset/tile_config.json]: Why the changes to tilesets?

File: data/json/items/software.json (hunk 1 above)
Comment (mugling) [inline data/json/items/software.json]: Fair enough

File: gfx/ChestHole16Tileset/tile_config.json (hunk 2 above)
Comment (codemime) [inline gfx/ChestHole16Tileset/tile_config.json]: The same reason as above.
